tests/unit/compat/mock.py

```diff
@@ -51,8 +51,7 @@ def _iterate_read_data(read_data):
             # newline that our naive format() added
             data_as_list[-1] = data_as_list[-1][:-1]
 
-        for line in data_as_list:
-            yield line
+        yield from data_as_list
```

Comment:
If you replace `for ... yield` by `yield from`, then it won't work with Python 2 anymore. You have to decide whether you actively want to break support for Python 2 in tests.